From 5e94e315119117898ce1a2798641848f61d796b1 Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Thu, 9 Jul 2020 07:58:44 +0000
Subject: [PATCH] configure.ac: define a path for profile data

There comes below error when use ccache 3.7.10 to compile python3
and check [1] for more details.
 | Python-3.8.3/Modules/_contextvarsmodule.c:43:1: error: source locations for function 'PyInit__contextvars' have changed, the profile data may be out of date [-Werror=coverage-mismatch]

That's because the logic for profile directory changes a little in
[2] after ccache upgrades to 3.7.10.

So define a profile directory path accordingly to fix the above error.

[1] https://github.com/ccache/ccache/issues/615
[2] https://github.com/ccache/ccache/commit/91a2954eb47b4a106e2be6cf611917b895108e35

Upstream-Status: Submitted [https://github.com/python/cpython/pull/21408]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 08fe397..bb15bda 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1332,8 +1332,8 @@ case $cc_basename in
         fi
         ;;
       *)
-        PGO_PROF_GEN_FLAG="-fprofile-generate"
-        PGO_PROF_USE_FLAG="-fprofile-use -fprofile-correction"
+        PGO_PROF_GEN_FLAG="-fprofile-generate=$(pwd)"
+        PGO_PROF_USE_FLAG="-fprofile-use=$(pwd) -fprofile-correction"
         LLVM_PROF_MERGER="true"
         LLVM_PROF_FILE=""
         ;;
-- 
2.26.2

